src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let tempClone = cloneObj(state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        tempClone = Object.assign({}, tempClone, action.extraData);
+        states.push(cloneObj(tempClone));
```

Comment:
why do you need to `cloneObj` in each case?
You repeat yourself 3 times
What if we do it after your  `switch / case` ?